src/serac/numerics/expr_template_impl.hpp

```diff
@@ -76,11 +76,15 @@ using enable_if_mfem_vec = std::enable_if_t<std::is_base_of_v<mfem::Vector, std:
 template <typename vec>
 auto index(vec&& v, const int idx)
 {
-  if constexpr (std::is_same_v<std::decay_t<vec>, mfem::HypreParVector>) {
+  constexpr bool vec_is_a_HypreParVector = std::is_same_v<std::decay_t<vec>, mfem::HypreParVector> ||
+                                           std::is_base_of_v<mfem::HypreParVector, std::decay_t<vec>>;
+
+  if constexpr (vec_is_a_HypreParVector) {
     return static_cast<const double*>(v)[idx];
   }
 
-  if constexpr (!std::is_same_v<std::decay_t<vec>, mfem::HypreParVector>) {
+  constexpr bool vec_isnt_a_HypreParVector = !vec_is_a_HypreParVector;
```

Comment:
Did this not work w/o pulling it out? I prefer leaving boolean values positive then use the `!` for readability.

Reply:
Either one should work.

I think this was one of my changes-- the motivation was that expressions like `std::is_same_v<std::decay_t<vec>, mfem::HypreParVector>`  are a little heavy on the type-trait stuff, so giving it a plain name was preferable. The thing I like about
```cpp
if (vec_isnt_a_HypreParVector) { ... }
```
is that it just reads like an english sentence.

---------
I wish C++ supported boolean operators on types, so we could just write
```cpp
if (vec != HypreParVector) { ... }
```